main/test/ca/uwaterloo/flix/library/Test.Exp.Struct.Put.flix

```diff
@@ -0,0 +1,136 @@
+        }
+
+    @test
+    def testStructPut03(): Bool =
+        region rc {
+            let s = new Put { _fstName = "Lucky", _lstName = "Luke", _age = 42, _cowboy = true } @ rc;
+            s€_age = 23;
+            s€_age == 23
+        }
+
+    @test
+    def testStructPut04(): Bool =
+        region rc {
+            let s = new Put { _fstName = "Lucky", _lstName = "Luke", _age = 42, _cowboy = true } @ rc;
+            s€_cowboy = false;
+            not s€_cowboy and s€_fstName == "Lucky" and s€_lstName == "Luke" and s€_age == 42
+        }
+
+    @test
+    def testStructMultiPut01(): Bool =
+        region rc {
+            let s = new Put { _fstName = "Lucky", _lstName = "Luke", _age = 42, _cowboy = true } @ rc;
+            s€_fstName = "Unlucky";
+            s€_lstName = "Uriel";
+            s€_age = 23;
+            s€_cowboy = false;
+            s€_fstName == "Unlucky" and s€_lstName == "Uriel" and s€_age == 23 and s€_cowboy == false
+        }
+
+    @test
+    def testStructMultiPut02(): Bool =
+        region rc {
+            let s = new Put { _fstName = "Lucky", _lstName = "Luke", _age = 42, _cowboy = true } @ rc;
+            s€_cowboy = false;
+            s€_age = 23;
+            s€_lstName = "Uriel";
+            s€_fstName = "Unlucky";
+            s€_fstName == "Unlucky" and s€_lstName == "Uriel" and s€_age == 23 and s€_cowboy == false
+        }
+
+    @test
+    def testStructMultiPut03(): Bool =
+        region rc {
+            let s = new Put { _fstName = "Lucky", _lstName = "Luke", _age = 42, _cowboy = true } @ rc;
+            s€_fstName = "Unlucky";
+            s€_fstName = "Lucky";
+            s€_lstName = "Uriel";
+            s€_lstName = "Luke";
+            s€_lstName == "Luke" and s€_fstName == "Lucky" and s€_age == 42 and s€_cowboy == true
+        }
+
+    @test
+    def testStructMultiPut04(): Bool =
+        region rc {
+            let s = new Put { _fstName = "Lucky", _lstName = "Luke", _age = 42, _cowboy = true } @ rc;
+            s€_lstName = "Uriel";
+            s€_fstName = "Unlucky";
+            s€_lstName = "Luke";
+            s€_fstName = "Lucky";
+            s€_fstName == "Lucky" and s€_lstName == "Luke" and s€_age == 42 and s€_cowboy == true
+        }
+
+/* Cannot test these while struct name is required to be mod name
+    struct Inner [r] {
+        x: Int32,
+        y: Int32
+    }
+
+    struct Outer [r] {
+        a: Inner[r]
+        b: Inner[r]
+    }
+
+    @test
+    def testNestedPut01(): Unit =
+        region rc {
+            let s = new Outer { a = new Inner { x = 1, y = 2}, b = new Inner { x = 1, y = 4 } } @ rc;
+            let _ax = s€a€x;
+            let _ay = s€a€y;
+            let _bx = s€b€x;
+            let _by = s€b€y;
+            ()
+        }
+
+    struct A [r] {
+        a: B
+    }
+
+    struct B [r] {
+        b: C
+    }
+
+    struct C [r] {
+        c: D
+    }
+
+    struct D [r] {
+        d: Int32
+    }
+
```

Comment:
Same comment about recursive and polymorphic structs.

Reply:
Would we be able to merge this and then I make another PR with recursive and polymorphic structs? 

This is just so that all my other PRs get tested with this at minimum